pytensor/graph/basic.py

```diff
@@ -907,6 +907,62 @@ def graph_inputs(
     yield from (r for r in ancestors(graphs, blockers) if r.owner is None)
 
 
+def required_graph_inputs(
+    graph: Variable[Any, Any] | Iterable[Variable[Any, Any]],
+) -> list[Variable[Any, Any]]:
+    """
+    Get the inputs into PyTensor variables
+
+    Parameters
+    ----------
+        graph: PyTensor `Variable` instances
+        Output `Variable` instances from which to search backward through
+        owners.
+
+    Returns
+    -------
+        List of tensor variables that are input nodes with no owner, in the order
+        found by a left-recursive depth-first search started at the nodes in `graphs`.
+
+    Examples
+    --------
+    >>> import pytensor as pt
+    >>> x=pt.vector('x')
+    >>> y=pt.constant('y')
+    >>> z = pt.mul(x*y)
+    >>> required_graph_inputs([a])
+    [[[ 0  1  2  3]
+      [ 4  5  6  7]
+      [ 8  9 10 11]]
+
+     [[12 13 14 15]
+      [16 17 18 19]
+      [20 21 22 23]]]
+
+
+    >>> pt.matrix_transpose(x).eval()
+    [[[ 0  4  8]
+      [ 1  5  9]
+      [ 2  6 10]
+      [ 3  7 11]]
+
+     [[12 16 20]
+      [13 17 21]
+      [14 18 22]
+      [15 19 23]]]
+    """
+    from pytensor.compile.sharedvalue import SharedVariable
+
+    if isinstance(graph, Variable):
+        graph = [graph]
+
+    return [
+        v
+        for v in graph_inputs(graph)
+        if isinstance(v, Variable) and not isinstance(v, Constant | SharedVariable)
+    ]
```

Comment:
Perhaps for consistency, it's better to return a generator. The user can call `list` on the output to convert it to a list if they wish to?

```suggestion
    return (
        v
        for v in graph_inputs(graph)
        if isinstance(v, Variable) and not isinstance(v, Constant | SharedVariable)
    )
```